src/sbt-test/sbtix/simple/sbtix.nix

```diff
@@ -20,8 +20,30 @@ rec {
     buildSbtProject = args@{repo, name, buildInputs ? [], sbtOptions ? "", ...}:
         stdenv.mkDerivation (rec {
             mvn = mkMavenRepo "${name}-repo" repo;
-            configurePhase = ''echo "externalResolvers in ThisBuild := Seq(\"nix\" at \"file://${mvn}\")" >> build.sbt'';
-            buildPhase = ''sbt compile ${sbtOptions}'';
+
+            #set environment variable to affect all SBT commands
+            SBT_OPTS = ''
+             -Dsbt.ivy.home=./.ivy2/
+             -Dsbt.boot.directory=./.sbt/boot/
+             -Dsbt.override.build.repos=true
+             -Dsbt.repository.config=./.sbtix_repos
+             ${sbtOptions}'';
+            
+            #eventually this repo list should only contain the nix repo
+            configurePhase = ''
```

Comment:
It might be cleaner to do this using [pkgs.writeText](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/trivial-builders.nix#L44) instead.